### PR TITLE
add codegen test for sigv4a EPR

### PR DESCRIPTION
**Pull request by Velfi**
exactly what it says on the tin. Let me know if we should have any other tests verifying the function of sigv4a auth schemes.

----

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecoratorTest.kt [hunk 1]
```diff
@@ -56,10 +62,113 @@ class SigV4AuthDecoratorTest {
...
+                        let config = $moduleUseName::Config::builder()
+                            .http_client(http_client.clone())
+                            .endpoint_url("https://example.com")
+                            .behavior_version_latest()
+                            .with_test_defaults()
+                            .build();
+                        let client = $moduleUseName::Client::from_conf(config);
+                        let _ = client.some_operation().something(#{ByteStream}::from_static(b"Hello, world!")).send().await;
+
+                        http_client.assert_requests_match(&[]);
```
Comment (ysaito1001) [inline aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecoratorTest.kt:158]: A bit concerned the use of `assert_requests_match` since the verification tries matching `Signature`, which is subject to change as we add new business metrics moving forward (new business metrics may be included in `x-amz-user-agent` by default in SDK codegen decorators, so it can affect a Kotlin test like this one). When new business metrics were introduced in a PR, that PR would probably fail this test as part of semver hazards checks (which is why we introduced `relaxed_validate` in [this PR](https://github.com/smithy-lang/smithy-rs/pull/3786) to work around the problem).

Long story short, to verify the expected output of the test, I'd probably get the `authorization` header and check if it contains `AWS4-ECDSA-P256-SHA256`.

**Review by landonxjames — APPROVED** [review]

File: aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecoratorTest.kt (hunk 1 above)
Comment (Velfi) [reply aws/sdk-codegen/src/test/kotlin/software/amazon/smithy/rustsdk/SigV4AuthDecoratorTest.kt:158]: OK, sounds good.

**Review by ysaito1001 — APPROVED** [review]
